1024pix/pix
# cy - refactored e2e tests (fixtures)

## :unicorn: Problem
We needed to refactor the Cypress tests to improve readability and maintainability, particularly in how fixtures were loaded across the test suite. The previous approach involved loading each fixture manually within the `beforeEach` block, leading to code duplication and decreased clarity.

## :robot: Solution
I refactored the Cypress test suite by introducing a more concise method to handle fixture loading. Instead of calling `cy.task("db:fixture", ...)` for each fixture separately, I grouped the fixtures into an array and used a loop to load them all at once. This reduces redundancy, improves readability, and makes future modifications easier (e.g., adding new fixtures).

The refactor only affects the way fixtures are loaded, without impacting the overall test functionality. This should help streamline future test maintenance and updates.

## :rainbow: Notes
- The fixtures array contains all the necessary fixtures and is processed in the `beforeEach` hook to ensure that they are loaded before each test.
- No changes were made to the actual test logic, only the setup phase for better readability.

## :100: How to test
1. Run the Cypress tests as usual.
2. Ensure that all tests relying on database fixtures run without issues and that the fixtures are loaded correctly for each test case.
3. Review the fixture-loading behavior to verify that it's functioning as expected across all test environments.
